src/main/java/com/google/api/codegen/gapic/GapicGeneratorApp.java

```diff
@@ -137,6 +149,22 @@ protected void process() throws Exception {
       }
     }
 
+    // Read the sample configs, if they are given, and convert them to protos.
+    SampleConfigProto sampleConfigProto = null;
+    List<String> sampleConfigFileNames = options.get(SAMPLE_CONFIG_FILES);
+    if (sampleConfigFileNames.size() > 0) {
+      ConfigSource configSource =
+          loadConfigFromFiles(
+              sampleConfigFileNames,
+              SampleConfigProto.getDescriptor().getFullName(),
+              SampleConfigProto.getDefaultInstance());
+
+      // TODO(hzyi): Verify this works for repeated fields as well
+      // TODO(hzyi): Allow users to put arbitrary top-level directives not
```

Comment:
Just a thought: if we agree gapic-config should have the same semantics (I think it should, but we should decide as a team) then this logic can be put in a function called from `loadConfigFromFiles`, which will ignore any top level YAML field that is not a top-level proto message field

Reply:
We can I think.